addons/account/models/chart_template.py

```diff
@@ -578,6 +582,131 @@ def _setup_utility_bank_accounts(self, template_code, company, template_data):
+            if existing_account := existing_accounts.get(template_id):
+                return existing_account
+
+            sign_comparator = '<' if int(foreign_tax_rep_line.get('factor_percent', 100)) < 0 else '>'
+            minimal_domain = [
+                ('company_id', '=', company.id),
+                ('account_id', '!=', False),
+                ('factor_percent', sign_comparator, 0),
+            ]
+            additional_domain = [
+                ('tax_id.type_tax_use', '=', type_tax_use),
+                ('tax_id.country_id', '=', company.account_fiscal_country_id.id),
+                ('tax_id', 'in', default_company_taxes.ids),
+            ]
+
+            # Trying to find an account being less restrictive on each iteration until the minimum acceptable is reached
+            # If nothing is found, don't fill it to avoid setting a wrong account
+            similar_repartition_line = None
+            while not similar_repartition_line and additional_domain:
+                search_domain = minimal_domain + additional_domain
+                similar_repartition_line = self.env['account.tax.repartition.line'].search(search_domain, limit=1)
+                additional_domain.pop()
+
+            similar_account_id = None
+            if similar_repartition_line:
+                local_tax_account = similar_repartition_line.account_id
+                similar_account_id = create_foreign_tax_account(local_tax_account, _("Foreign tax account (%s)", country.code)).id
+
+            existing_accounts[template_id] = similar_account_id
+            return existing_accounts[template_id]
+
+        # Retrieve the required l10n datas
+        chart_template_code = self._guess_chart_template(country=country)
+        tax_groups = self._get_chart_template_data(chart_template_code)['account.tax.group']
+        taxes = self._get_chart_template_data(chart_template_code)['account.tax']
+
+        # try to replace tax group default accounts by a proper one or clean the given xml_id as it doesn't exist
+        # in the company's chart template
+        property_name = (
+            ('tax_payable_account_id', _("Foreign tax account payable (%s)", country.code)),
+            ('tax_receivable_account_id', _("Foreign tax account receivable (%s)", country.code)),
+            ('advance_tax_payment_account_id', _("Foreign tax account advance payment (%s)", country.code)),
+        )
+        property_company = self.env['ir.property'].with_company(company)
+        for account_field, account_name in property_name:
+            property_account_field = f"property_{account_field}"
+            default_acc = property_company._get(property_account_field, 'account.tax.group')
+            foreign_account_id = create_foreign_tax_account(default_acc, account_name).id if default_acc else False
+            for tax_group in tax_groups.values():
+                tax_group[account_field] = foreign_account_id
```

Comment:
Are those still properties? :doubt:
https://github.com/odoo/odoo/blob/f48147587956fc30773f0dbac5086e9f9bc0c970/addons/account/models/account_tax.py#L29-L43

Reply:
I should add some more tests for that.

Reply:
Looks like you are still using company properties, which is obviously wrong :wink:

Reply:
That's why the last commit is "WOIP" :grin:

Reply:
Me no read